code/entities/chair/Armrest.cs

```diff
@@ -0,0 +1,129 @@
+﻿using Sandbox;
+using System;
+
+namespace Cinema;
+
+public partial class Armrest : BaseNetworkable
+{
+    public enum States : int
+    {
+        Lowered,
+        Raised
+    }
+
+    public enum Sides : int
+    {
+        Left,
+        Right
+    }
+
+    [Net]
+    public CinemaChair Chair { get; set; }
+    [Net]
+    public ModelEntity HeldEntity { get; set; }
+
+    [Net]
+    public Sides Side { get; set; }
+
+    public States State => Chair.GetAnimParameterBool(ArmrestToggleParameter)
+    ? States.Raised
+    : States.Lowered;
+
+    public string CupholderBoneName => Side switch
+    {
+        Sides.Left => "cupholder_l",
+        _ => "cupholder_r"
+    };
+
+    private Transform GetCupholderTransform()
+        => Chair.GetBoneTransform(CupholderBoneName);
+
+    public string ArmrestToggleParameter => Side switch
+    {
+        Sides.Left => "toggle_left_armrest",
+        _ => "toggle_right_armrest"
+    };
+
+    public void HandleAnimTag(string tag)
+    {
+        if (tag.Contains("launch"))
+        {
+            // Default to launching HeldEntity back and up a bit.
+            var launchDirection = (Chair.Rotation.Backward + Chair.Rotation.Up).Normal;
+            LaunchHeldEntity(launchDirection, 250f, 0.2f);
+        }
+    }
+
+    public void Toggle()
+    {
+        var originalValue = Chair.GetAnimParameterBool(ArmrestToggleParameter);
+        Chair.SetAnimParameter(ArmrestToggleParameter, !originalValue);
+    }
+
+    /// <summary>
+    /// Will set the entity contained within the cupholder of this armrest. Will
+    /// return without doing anything if the <c>State</c> is not lowered. Will 
+    /// launch any entity that is already held.
+    /// </summary>
+    /// <param name="entity">The entity that shall be held in the cupholder.</param>
+    public bool TryHoldEntity(ModelEntity entity)
+    {
+        if (State != States.Lowered)
+        {
+            return false;
+        }
+        if (HeldEntity != null)
+        {
+            var sideVec = (Vector3)Vector2.Random * 0.5f;
+            // Just pop the held entity up and out out of the cupholder.
+            LaunchHeldEntity(Chair.Rotation.Up + sideVec, 150f);
```

Comment:
This could lead to a potential race condition if the `LaunchHeldEntity` doesn't execute immediately, or the default parameter `disableCollisionTime` is ever changed from 0. I'm not sure if the former would ever happen in C#, so maybe we can ignore that.

Either way, this isn't blocking, just a warning of a potential bug in the future if the `LaunchHeldEntity` code changes very much.